### PR TITLE
Preserve focus styles while the component is focused.

### LINKED ISSUE #8507: Checkbox, switch, and radio buttons lose focus styles.
## Description 
Currently, when I focus on those components using the keyboard, the focus style disappears right after an interaction. For example, pressing the spacebar on a checkbox will check the checkbox but the checkbox will lose focus style. I noticed that the class for the focus styles igx-xxx--focus disappears right after the interaction and does not appear until the element is focused again with the keyboard.

Also, the arrow navigation between the radio buttons group does not change the focus.

 * igniteui-angular version: all
 * browser: all

## Steps to reproduce  

1. Open dev demos
2. got to Forms ([link](http://localhost:4200/inputs))
3. use your keyboard to focus on a checkbox
4. press spacebar to check
5. observe that the styles for the focus are gone
6. to make them visible again prest TAB and SHIFT+TAB

## Result  
The class for focus style disappears right after the element is pressed.

## Expected result  
The focus style should persist on an element until the element is no longer on focus.

### PR DESCRIPTION
**Pull request by desig9stein**
Closes #8507

### Additional information (check all that apply):
 - [x] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/checkbox/checkbox.component.ts [hunk 1]
```diff
@@ -369,34 +381,31 @@ export class IgxCheckboxComponent implements ControlValueAccessor, EditorProvide
...
+
     /** @hidden @internal */
-    public _onLabelClick(event) {
-        // We use a span element as a placeholder label
-        // in place of the native label, we need to emit
-        // the change event separately here alongside
-        // the click event emitted on click
+    public _onLabelClick() {
         this.toggle();
+        this.nativeCheckbox.nativeElement.focus();
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/checkbox/checkbox.component.ts:394]: The toggle method already calls the focus on the native checkbox input.

File: projects/igniteui-angular/src/lib/radio/radio.component.ts [hunk 2]
```diff
@@ -314,6 +315,16 @@ export class IgxRadioComponent implements ControlValueAccessor, EditorProvider {
 
     constructor(private cdr: ChangeDetectorRef) { }
 
+    /**
+     * @hidden
+     * @internal
+     */
+    @HostListener('keyup', ['$event'])
+    public onKeydown(event: KeyboardEvent) {
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/radio/radio.component.ts:323]:
```suggestion
    public onKeyUp(event: KeyboardEvent) {
```

File: projects/igniteui-angular/src/lib/switch/switch.component.ts [hunk 3]
```diff
@@ -255,6 +256,16 @@ export class IgxSwitchComponent implements ControlValueAccessor, EditorProvider
      * @internal
      */
     private _onChangeCallback: (_: any) => void = noop;
+    /**
+     * @hidden
+     * @internal
+     */
+    @HostListener('keyup', ['$event'])
+    public onKeydown(event: KeyboardEvent) {
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/switch/switch.component.ts:264]:
```suggestion
    public onKeyUp(event: KeyboardEvent) {
```

File: projects/igniteui-angular/src/lib/checkbox/checkbox.component.ts [hunk 4]
```diff
@@ -328,6 +329,15 @@ export class IgxCheckboxComponent implements ControlValueAccessor, EditorProvide
      * @hidden
      */
     private _onChangeCallback: (_: any) => void = noop;
+    /**
+     * @hidden
+     * @internal
+     */
+    @HostListener('keyup', ['$event'])
+    public onKeydown(event: KeyboardEvent) {
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/checkbox/checkbox.component.ts:337]:
```suggestion
    public onKeyUp(event: KeyboardEvent) {
```

**Review by simeonoff — CHANGES_REQUESTED** [review]

**Review by simeonoff — DISMISSED** [review]

**Review by kdinev — APPROVED** [review]

**Pull request merged**